MekHQ/src/mekhq/gui/CampaignGUI.java

```diff
@@ -1611,7 +1613,11 @@ public void refitUnit(Refit r, boolean selectModelName) {
                         + tech.getMinutesLeft() + "/" + TimePerDay
                         + " minutes";
                 techHash.put(name, tech);
-                techList.add(name);
+                if(tech.isRightTechTypeFor(r)) {
+                    techList.add(lastRightTech++, name);
```

Comment:
If this variable is removed and replaced with a 0, then you get the best techs to refit with first, while any other techs are sorted descending.

Alternatively, you could swap Line 1590 to:
```List<Person> techs = getCampaign().getTechs(false, null, true, true);```
And keep the lastRightTech, which would have the best of the proper type first, then the best of any other type, sorted by their time left.

Both changes allow one to remove the PrimaryRole sort, as why do we need to group the others by role?

Reply:
The idea was minimal change. Inserting in this order maintains the same order the user will see in the rest of the list, just with proper techs promoted to the front. I like the idea of removing PrimaryRole sort though. Back to the salt mines ...